syntax = "proto3";
package userservice;

service UserService {
  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse) {}
  rpc AuthenticateUser(LoginUserRequest) returns (LoginUserResponse) {}
}

message CreateUserRequest {
    string username=1;
    string password=2;
    string email=3;
}

message CreateUserResponse {
    string user_id=1;
    string username=2;
    string email=3;
}

message LoginUserRequest {
    string username=1;
    string password=2;
}


message LoginUserResponse {
    string access_token = 1;
    string token_type = 2;
}